packages/main/src/plugin/provider-registry.spec.ts

```diff
@@ -444,6 +449,172 @@ describe('should send events when starting a container connection', async () =>
   });
 });
 
+describe('when auto-starting a container connection', async () => {
+  let connection: ProviderContainerConnectionInfo;
+  let provider: Provider;
+  let containerProviderConnection: ContainerProviderConnection;
+
+  beforeEach(() => {
+    provider = providerRegistry.createProvider('id', 'name', {
+      id: 'internal',
+      name: 'internal',
+      status: 'installed',
+    });
+    connection = {
+      name: 'connection',
+      displayName: 'connection',
+      type: 'docker',
+      endpoint: {
+        socketPath: '/endpoint1.sock',
+      },
+      status: 'started',
+      vmType: {
+        id: 'libkrun',
+        name: 'libkrun',
+      },
+    };
+
+    containerProviderConnection = {
```

Comment:
If connection and containerProviderConnection are never updated, they should not be part of the beforeEach, they should be defined outside as they are constants